src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -390,6 +390,54 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
         }
 
         public static bool CheckConstraints(
```

Comment:
Since `elementLocations` is flatted among all nested tuples an `offset` is needed to map to associate diagnostics with their corresponding element's position.